src/Logger/LoggerFactory.php

```diff
@@ -72,121 +74,109 @@ public function __construct(
-        return new SummaryFileLogger(
-            $output,
-            $location,
-            $this->metricsCalculator,
-            $this->filesystem,
-            $isDebugVerbosity,
-            $this->debugMode
-        );
+        ?string $filePath
+    ): ?FileLogger {
+        return $filePath === null
+            ? null
+            : new FileLogger(
+                $output,
+                $filePath,
+                $this->filesystem,
+                new SummaryFileLogger($this->metricsCalculator)
+            )
+        ;
     }
 
     private function createDebugLogger(
         OutputInterface $output,
-        string $location,
-        bool $isDebugVerbosity
-    ): DebugFileLogger {
-        return new DebugFileLogger(
-            $output,
-            $location,
-            $this->metricsCalculator,
-            $this->filesystem,
-            $isDebugVerbosity,
-            $this->debugMode,
-            $this->onlyCoveredCode
-        );
+        ?string $filePath
+    ): ?FileLogger {
+        return $filePath === null
+            ? null
+            : new FileLogger(
+                $output,
+                $filePath,
+                $this->filesystem,
+                new DebugFileLogger($this->metricsCalculator, $this->onlyCoveredCode)
+            )
+        ;
     }
 
     private function createPerMutatorLogger(
         OutputInterface $output,
-        string $location,
-        bool $isDebugVerbosity
-    ): PerMutatorLogger {
-        return new PerMutatorLogger(
-            $output,
-            $location,
-            $this->metricsCalculator,
-            $this->filesystem,
-            $isDebugVerbosity,
-            $this->debugMode
-        );
+        ?string $filePath
+    ): ?FileLogger {
+        return $filePath === null
+            ? null
+            : new FileLogger(
+                $output,
+                $filePath,
+                $this->filesystem,
+                new PerMutatorLogger($this->metricsCalculator)
+            )
+        ;
     }
 
-    private function createBadgeLogger(OutputInterface $output, string $branch): BadgeLogger
+    private function createBadgeLogger(OutputInterface $output, ?Badge $badge): ?BadgeLogger
     {
-        return new BadgeLogger(
-            $output,
-            new ChainBuildContextResolver(new TravisCiResolver()),
-            new StrykerApiKeyResolver(),
-            new StrykerDashboardClient(
-                new JsonClient(),
-                new ConsoleLogger($output)
-            ),
-            $this->metricsCalculator,
-            $branch
-        );
+        return $badge === null
+            ? null
+            : new BadgeLogger(
+                $output,
+                new ChainBuildContextResolver(new TravisCiResolver()),
+                new StrykerApiKeyResolver(),
+                new StrykerDashboardClient(
+                    new JsonClient(),
+                    new ConsoleLogger($output)
```

Comment:
I wish BadgeLogger would know these details itself. Say within a `::build()` method.

Reply:
I'm still uncertain about the fate of `LoggerFactory` and attempted something in #1094. So I prefer to not touch that part in this PR